noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/components/private_call_data_validator.nr

```diff
@@ -121,17 +121,6 @@ impl PrivateCallDataValidator {
         let call_context = public_inputs.call_context;
         assert(call_context.is_delegate_call == false, "Users cannot make a delegatecall");
         assert(call_context.is_static_call == false, "Users cannot make a static call");
-
-        let min_revertible_side_effect_counter = public_inputs.min_revertible_side_effect_counter;
-        let first_revertible_index = find_first_revertible_private_call_request_index(public_inputs);
-        // No need to check that the min_revertible_side_effect_counter falls in the counter range of the private call.
-        // It is valid as long as it does not fall in the middle of any nested call.
-        validate_split_ranges(
```

Comment:
Moved it down to `validate_private_call_requests`, which is run by both init and inner. Because now we can set `min_revertible_side_effect_counter` from any private function, not just the entrypoint. So we need to perform this check when the counter is set.